ruma-events-macros/src/event_enum.rs

```diff
@@ -572,85 +528,93 @@ mod kw {
     syn::custom_keyword!(events);
 }
 
-pub enum EnumKind {
+pub enum EventKindVariation {
+    Full,
+    Stub,
+    Stripped,
+    Redacted,
+    RedactedStub,
+    RedactedStripped,
+}
+
+pub enum EventKind {
     Basic(Ident),
     Ephemeral(Ident),
     Message(Ident),
     State(Ident),
     ToDevice(Ident),
 }
 
-impl EnumKind {
-    /// Return the span of this ident.
-    fn span(&self) -> Span {
-        match &self {
-            Self::Basic(i)
-            | Self::Ephemeral(i)
-            | Self::Message(i)
-            | Self::State(i)
-            | Self::ToDevice(i) => i.span(),
-        }
-    }
-    /// `[Kind]Event`
-    fn to_full_event_struct(&self) -> Ident {
-        match &self {
-            Self::Basic(i)
-            | Self::Ephemeral(i)
-            | Self::Message(i)
-            | Self::State(i)
-            | Self::ToDevice(i) => format_ident!("{}Event", i),
+impl EventKind {
+    fn is_state(&self) -> bool {
+        if let Self::State(_) = self {
```

Comment:
You can use `matches!` from the matches crate here.

Reply:
Do we have a minimum rust version we support? Or can I use the `std::matches!` stabel since 1.42 or do they back port stuff or how does that work?

Reply:
I'd say we should use the matches crate for now. Ideally I'd like our MSRV to be 1.40 for a while, but ruma-signatures depends on ring, which has an MSRV of 'latest stable', so we can at max apply this to the other crates.

Reply:
Opened an issue: #122

Reply:
Cool!